### PR TITLE
hdf5-mpi: Fix build for Linux

### PR DESCRIPTION
**Pull request by ax3l**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

~~Note: the `autogen.sh` part of this PR is proposed for `homebrew-core`: https://github.com/Homebrew/homebrew-core/pull/57051~~

### CODE REVIEW

File: Formula/hdf5-mpi.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def install
               "settingsdir=$(libdir)",
               "settingsdir=#{pkgshare}"
 
-    system "autoreconf", "-fiv"
```
Comment (iMichka) [inline Formula/hdf5-mpi.rb:34]:
```
if OS.mac?
  system "autoreconf", "-fiv"
else
  system "./autogen.sh"
end
```

But why is this different on Mac and Linux?

Comment (ax3l) [reply Formula/hdf5-mpi.rb:34]: We should actually be using `./autogen.sh` on both platforms, but due to some hard-coded logic bugs we have to fall back to calling atuoreconf our selves on macOS, as far as I can see what is going on here.

File: Formula/hdf5-mpi.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@ def install
...
+    system "./autogen.sh"
```
Comment (ax3l) [inline Formula/hdf5-mpi.rb:34]: @iMichka suggests:
```suggestion
    if OS.mac?
        system "autoreconf", "-fiv"
    else
        system "./autogen.sh"
    end
```

**Comment by iMichka** [comment]
Thanks @ax3l !

**Pull request merged**